QNX: GCredentials via SCM_CREDS/struct cmsgcred, as on FreeBSD (GDBus EXTERNAL)

Upstream-Status: QNX-specific

--- a/gio/gcredentialsprivate.h
+++ b/gio/gcredentialsprivate.h
@@ -117,7 +117,8 @@
 #elif defined(__FreeBSD__)                                  || \
       defined(__FreeBSD_kernel__) /* Debian GNU/kFreeBSD */ || \
       defined(__GNU__)            /* GNU Hurd */            || \
-      defined(__DragonFly__)      /* DragonFly BSD */
+      defined(__DragonFly__)      /* DragonFly BSD */ || \
+      defined(__QNX__)            /* QNX: FreeBSD's socket layer (io-sock) */
 #define G_CREDENTIALS_SUPPORTED 1
 #define G_CREDENTIALS_USE_FREEBSD_CMSGCRED 1
 #define G_CREDENTIALS_NATIVE_TYPE G_CREDENTIALS_TYPE_FREEBSD_CMSGCRED
